src/main/MainScreen.js

```diff
@@ -20,7 +21,17 @@ export default class MainScreen extends React.Component {
   }
 
   render() {
-    const { streamlistOpened, doNarrow } = this.props;
+    const { streamlistOpened, doNarrow, narrow, subscriptions } = this.props;
+    let color = BRAND_COLOR;
+    let i = 0;
```

Comment:
You can simplify this by using a functional programming style:
```
if (narrow.length != 0 && narrow[0].operator === 'stream') {
  color = subscriptions.find((sub) => narrow[0].operand === sub.name);
}
```

The find function returns the first value in an array that causes the predicate (boolean function) you specify to return `true`.  If nothing is found it returns `undefined`